res/controllers/DJ TechTools-MIDI Fighter Twister-scripts.js

```diff
@@ -54,6 +54,157 @@ var MidiFighterTwister;
     components.Button.prototype.on = engine.getSetting("defColor");
     components.Button.prototype.off = engine.getSetting("relColor");
 
+    class StemDeck extends components.Deck {
+        midiModifier(value) {
+            // All midi values are for the left deck, we only modify them if
+            // we're constructing the right hand deck.
+            if (this.deckNumbers[0] === 1) {
+                return value;
+            }
+
+            // Even controls (ie. the first row) are mirrored to the far right
+            // row, three rows over.
+            if (value % 2 === 0) {
+                return value + 3;
+            }
+            // Odd controls (ie. the second row) are mirrored to the other inner
+            // row, one row over.
+            return value + 1;
+        }
+
+        constructor(deckNums) {
+            super(deckNums);
+
+            this.volume = [];
+            this.mute = [];
+            this.effect = [];
+            this.effectEnabled = [];
+            this.effectVolume = [];
+            this.effectReset = [];
+            for (let i = 0; i < 4; i++) {
+                this.volume[i] = new components.Encoder({
+                    group: `[Channel${deckNums[0]}_Stem${i + 1}]`,
+                    midi: [0xB0, this.midiModifier(0x20 + (4 * i))],
+                    key: "volume",
+                });
+                // TODO: change these into mute or reset buttons based on a
+                // setting?
+                this.mute[i] = new components.Button({
+                    group: `[Channel${deckNums[0]}_Stem${i + 1}]`,
+                    midi: [0xB1, this.midiModifier(0x20 + (4 * i))],
+                    key: "mute",
+                    colorKey: "color",
+                    colorMapper: new ColorMapper({
+                        0x0040f0: 1,   // Blue
+                        0x3ec8e6: 32,  // Celeste
+                        0x008080: 40,  // Teal
+                        0x2fb340: 50,  // Green
+                        0x3fff58: 60,  // Lime
+                        0xe9c600: 66,  // Yellow
+                        0xf07800: 74,  // Orange
+                        0xfd4a4a: 85,  // Red (was being mapped to orange, perceptually closer to red)
+                        0xb90908: 85,  // Red
+                        0xff00ff: 100, // Fuscia
+                        0xa400c0: 108, // Purple
+                    }),
+                    type: components.Button.prototype.types.toggle,
+                    outValueScale: function(_value) {
+                        return this.colorMapper.getValueForNearestColor(engine.getParameter(this.group, this.colorKey));
+                    },
```

Comment:
you're customizing `outValueScale`, but this will only ever get called by `output` (in the default impl). You're overwriting `output` though, so this looks like dead code to me?